### PR TITLE
refactor/rename-filter-subscription

### PR DESCRIPTION
**Pull request by decanus**
@oskarth here is a renaming, I don't know if the name is too long but it seems more explanatory.

### CODE REVIEW

File: tests/v2/test_waku_filter.nim [hunk 1]
```diff
@@ -11,7 +11,7 @@ import
   libp2p/multistream,
   libp2p/transports/transport,
   libp2p/transports/tcptransport,
-  ../../waku/protocol/v2/[waku_filter, filter],
+  ../../waku/protocol/v2/[waku_filter, message_notifier],
```
Comment (oskarth) [inline tests/v2/test_waku_filter.nim:14]: does this belong in protocol?

**Review by oskarth — COMMENTED** [review]
hm, isn't subscription already used for libp2p and protocols? semantics aside, not sure if it is that much better than filter :P

File: waku/protocol/v2/filter.nim [hunk 2]
```diff
@@ -1,37 +0,0 @@
-import
```
Comment (oskarth) [inline waku/protocol/v2/filter.nim:1]: Strange that this doesnt show up as a rename of file

File: tests/v2/test_waku_filter.nim (hunk 1 above)
Comment (decanus) [reply tests/v2/test_waku_filter.nim:14]: If we put it in node, we get weird import cycles I believe. Not sure tho

File: waku/protocol/v2/filter.nim (hunk 2 above)
Comment (decanus) [reply waku/protocol/v2/filter.nim:1]: Git!

**Review by oskarth — APPROVED** [review]

**Review by kdeme — APPROVED** [review]

**Pull request merged**